src/wasm.h

```diff
@@ -86,6 +86,10 @@ struct Name : public cashew::IString {
   static Name fromInt(size_t i) {
     return cashew::IString(std::to_string(i).c_str(), false);
   }
+
+  bool hasSubstring(cashew::IString substring) {
```

Comment:
I think it'd be useful to clarify what's convenient to have in `wasm.h` and why. From my perspective the file is too large and all-encompassing, and has too much implementation baked in to the header.

Would you be opposed to a separate PR to move function implementations into `wasm.cpp`?

For the scope of this PR, I'll just move `Name` to its own file.